### PR TITLE
[bugfix] Fix slow node crush in hive predictor

### PR DESCRIPTION
**Pull request by dawn310826**

### CODE REVIEW

File: easy_rec/python/utils/hive_utils.py [hunk 1]
```diff
@@ -48,11 +48,11 @@ def gen_sql(self):
     str_fields = ','.join(fields)
     if not part:
       sql += """
-    where hash(concat({}))%{}={}
+    where abs(hash(concat({})))%{}={}
     """.format(str_fields, self.task_num, self.task_index)
     else:
       sql += """
-    where {} and hash(concat({}))%{}={}
+    where {} and abs(hash(concat({})))%{}={}
```
Comment (tiankongdeguiji) [inline easy_rec/python/utils/hive_utils.py:55]: where CAST(ROUND(rand() * self.task_num) AS BIGINT) = self.task_index

File: easy_rec/python/inference/predictor.py [hunk 2]
```diff
@@ -903,7 +904,26 @@ def _get_reserve_vals(self, reserved_cols, output_cols, all_vals, outputs):
...
     return reserve_vals
 
-  def load_to_table(self, output_path):
+  def _check_worker_success(self, slice_num):
+    while (True):
+      for id in range(slice_num):
+        res_path = os.path.join(self._hdfs_path, 'SUCCESS-%s' % id)
+        if not gfile.Exists(res_path):
+          time.sleep(10)
+          self._check_worker_success(slice_num)
```
Comment (tiankongdeguiji) [inline easy_rec/python/inference/predictor.py:913]: too much recursion

**Review by tiankongdeguiji — CHANGES_REQUESTED** [review]

File: easy_rec/python/utils/hive_utils.py (hunk 1 above)
Comment (dawn310826) [reply easy_rec/python/utils/hive_utils.py:55]: where CAST(rand() * self.task_num) AS BIGINT) = self.task_index
drop ROUND

File: easy_rec/python/inference/predictor.py (hunk 2 above)
Comment (dawn310826) [reply easy_rec/python/inference/predictor.py:913]: done

**Review by tiankongdeguiji — APPROVED** [review]
LGTM

**Pull request merged**